src/CLICommand/cli.ts

```diff
@@ -1,23 +1,33 @@
-export interface ParsedArguments {
+import { CommanderError } from 'commander';
+
+export interface ParsedParameters {
 	// TODO: make parameterName to have type ParameterName
 	// eslint-disable-next-line @typescript-eslint/no-explicit-any
 	[parameterName: string /** ParameterName */]: any;
 }
+
 /**
  * @name CliApiObject
  * abstract class intended to help encapsulate the cli api object
  * also used for mocking in tests
  */
 export abstract class CliApiObject {
 	abstract arguments(names: string): CliApiObject;
-	abstract action(action: (options: ParsedArguments) => Promise<void>): CliApiObject;
+	abstract action(action: ActionCallback): CliApiObject;
 	abstract option(aliases: string, description: string, defaultValue?: string | boolean): CliApiObject;
 	abstract requiredOption(aliases: string, description: string, defaultValue?: string | boolean): CliApiObject;
 	abstract command(commandName: string): CliApiObject;
 	abstract parse(...args: [program: CliApiObject] | [argv: string[]]): void;
 	abstract addHelpCommand(addHelp: boolean): void;
-	abstract opts(): ParsedArguments;
+	abstract opts(): ParsedParameters;
 	abstract name(name: string): CliApiObject;
 	abstract usage(usage: string): CliApiObject;
 	abstract outputHelp(): void;
+	abstract exitOverride(callback?: (commanderError: CommanderError) => void): CliApiObject;
 }
+
+export type ActionCallback = (options: ParsedParameters) => ActionReturnType; // commander action callback
+
+export type AsyncActionCallback = (options: ParsedParameters) => Promise<ActionReturnType>;
+
+export type ActionReturnType = number | void;
```

Comment:
Consider:
```
alias ExitCode = number;
```
and use here.

Reply:
There: c91c80b8cdee17fec997d9842f3d1b4a52e2c817